### PR TITLE
[api][SIMS #407]CAR SIN Validation - closing DB connections properly

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
Found and fixed an issue on DB connections handling where the connection will remain open till the connection pool reaches the limit currently configured to 10.

### CODE REVIEW

File: sources/packages/api/src/cra-integration/cra-personal-verification.service.ts [hunk 1]
```diff
@@ -106,7 +106,7 @@ export class CRAPersonalVerificationService {
     // Executes the processing of each file in parallel.
     const filesProcess = files.map((file) => this.processResponse(file));
     // Waits for all the parallel processes to be finished.
-    return await Promise.all(filesProcess);
+    return Promise.all(filesProcess);
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/cra-integration/cra-personal-verification.service.ts:109]: Recommendation from Sonar Cloud.

File: sources/packages/api/src/services/student/student.service.ts [hunk 2]
```diff
@@ -148,7 +148,7 @@ export class StudentService extends RecordDataModelService<Student> {
    * @returns Students pending SIN validation.
    */
   async getStudentsPendingSinValidation(): Promise<Student[]> {
-    return await this.repo.find({ validSIN: null });
+    return this.repo.find({ validSIN: null });
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/services/student/student.service.ts:151]: Recommendation from Sonar Cloud.

File: sources/packages/api/src/services/sequence-control/sequence-control.service.ts [hunk 3]
```diff
@@ -78,13 +78,14 @@ export class SequenceControlService extends RecordDataModelService<SequenceContr
...
-      queryRunner.manager.save(sequenceRecord);
+      await queryRunner.manager.save(sequenceRecord);
       await queryRunner.commitTransaction();
     } catch (error) {
-      this.logger.error(`Error while executing the process ${error}`);
       this.logger.error("Executing sequence number rollback...");
       await queryRunner.rollbackTransaction();
       throw error;
+    } finally {
+      await queryRunner.release();
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/services/sequence-control/sequence-control.service.ts:88]: The real problem causing the issue.

File: sources/packages/api/src/cra-integration/cra-personal-verification.service.ts (hunk 1 above)
Comment (ann-aot) [reply sources/packages/api/src/cra-integration/cra-personal-verification.service.ts:109]: oh..but await is need for async right.Okay

Comment (andrewsignori-aot) [reply sources/packages/api/src/cra-integration/cra-personal-verification.service.ts:109]: Yes, it is, but not at this point. The consumer of this method will await this.

**Review by ann-aot — APPROVED** [review]

**Pull request merged**